### PR TITLE
Automatic notification reminders for interesting sessions

### PR DESCRIPTION
**Pull request by gewfy**
Asks the user if they want notification reminders  for sessions marked as interested

### Session card states
| For hosts | For users |
|-|-|
| image image | image image | 

### Session modal 
| For hosts | For users |
|-|-|
| image | image |

### Prompt
| Marking as interested | Asking for permissions |
|-|-|
| image | image |

### Other
| Updated setting | Interested component |
|-|-|
| image | image |

### CODE REVIEW

File: client/src/lib/notifications/hooks/useReminderNotificationsSetting.spec.ts [hunk 1]
```diff
@@ -112,57 +113,79 @@ describe('useNotificationSetting', () => {
...
       mockRequestPermission.mockResolvedValueOnce(undefined);
       useUserState.setState({
         user: {uid: 'some-uid'} as FirebaseAuthTypes.User,
       });
 
-      const {result} = renderHook(() => useNotificationSetting());
+      const {result} = renderHook(() => useReminderNotificationsSetting());
 
-      await result.current.setNotificationsEnabled(true);
+      await result.current.setRemindersEnabled(true);
```
Comment (swemail) [inline client/src/lib/notifications/hooks/useReminderNotificationsSetting.spec.ts:155]: Don't you get `act` warnings when doing this outside `act`?

**Review by swemail — APPROVED** [review]
Nice nice nice!

File: client/src/lib/notifications/hooks/useReminderNotificationsSetting.spec.ts (hunk 1 above)
Comment (gewfy) [reply client/src/lib/notifications/hooks/useReminderNotificationsSetting.spec.ts:155]: Hmmm maybe I do... will check!

Comment (gewfy) [reply client/src/lib/notifications/hooks/useReminderNotificationsSetting.spec.ts:155]: I do... I get a ton of them :D

**Pull request merged**